info-compiler/src/main/java/com/google/sps/webcrawler/RelevancyChecker.java

```diff
@@ -50,20 +50,26 @@ public RelevancyChecker() throws IOException {
 
   /**
    * Checks whether the {@code newsArticle} is relevant to the {@code candidateName} of interest.
-   * Defines relevancy as the salience of {@code candidateName} in the content, and defines
-   * sufficient relevancy with {@code SALIENCE_THRESHOLD}.
+   * Defines relevancy as the salience of {@code candidateName} and {@code partyName} in the
+   * content both being bigger than their respective threshold. If {@code partyName} is null,
+   * skips the salience checking for {@code partyName} and determines relevancy solely by looking
+   * at the salience of {@code candidateName}.
    */
-  public boolean isRelevant(NewsArticle newsArticle, String candidateName) {
-    double salience = computeSalienceOfName(newsArticle.getContent(), candidateName);
-    return salience >= SALIENCE_THRESHOLD;
+  public boolean isRelevant(NewsArticle newsArticle, String candidateName, String partyName) {
+    double candidateNameSalience = computeSalienceOfName(newsArticle.getContent(), candidateName);
+    double partyNameSalience =
+        (partyName == null)
```

Comment:
is partyName == null the same as "NO PARTY AFFILIATION"?

Reply:
No, this is for when the party affiliation field doesn't exist in the query response. However, all responses I have seen provide the party affiliation field. I can remove this check!

Reply:
Oh I was going to say to expand this check since it looks like there are several types of probably non-relevant parties which may skew the results: "NON PARTISAN", "NO PARTY AFFILIATION", "NONPARTISAN"
https://voter-central.ue.r.appspot.com/directory.html?address=default&listAllElections=true

Reply:
Oh that makes sense! Thank you I'll check for additional conditions.